pkg/diffs/diffs.go

```diff
@@ -97,27 +98,24 @@ func GetChangedPresubmits(prowMasterConfig, prowPRConfig *prowconfig.Config, log
 	masterJobs := getJobsByRepoAndName(prowMasterConfig.JobConfig.PresubmitsStatic)
 	for repo, jobs := range prowPRConfig.JobConfig.PresubmitsStatic {
 		for _, job := range jobs {
-			masterJob := masterJobs[repo][job.Name]
-			logFields := logrus.Fields{logRepo: repo, logJobName: job.Name}
-
-			if job.Agent == string(pjapi.KubernetesAgent) {
-				// If the agent was changed and is a kubernetes agent, just choose the job for rehearse.
-				if masterJob.Agent != job.Agent {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if !equality.Semantic.DeepEqual(masterJob.Spec, job.Spec) {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if masterJob.Optional && !job.Optional {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if !masterJob.AlwaysRun && job.AlwaysRun {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if masterJob.Cluster != job.Cluster {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
+			var why []string
```

Comment:
IMHO variables should describe what they contain, not what questions can be asked to which the content of the var is the correct answer, so something like `reasons` for the var name and log key makes IMHO more sense

Reply:
(as in: relation var name var content should be an equasion)

Reply:
:+1: